tools/c7n_gcp/tests/test_notify.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2015-2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from googleapiclient.errors import HttpError
+from gcp_common import BaseTest
+
+
+class NotifyTest(BaseTest):
+
+    def test_notify_schema_validate(self):
+        factory = self.replay_flight_data('gcpnotifyvalidation')
+        p = self.load_policy({
+            'name': 'test-notify-for-gcpfunction',
+            'resource': 'gcp.function',
+            'actions': [
+                {'type': 'notify',
+                 'template': 'default',
+                 'priority_header': '2',
+                 'subject': 'testing notify action',
+                 'to': ['user@domain.com'],
+                 'transport':
+                     {'type': 'pubsub',
+                      'topic': 'testtopic'}
+                 }
+            ]}, validate=True, session_factory=factory)
+        self.assertTrue(p)
+
+    def test_pubsub_notify(self):
```

Comment:
there's alot of extraneous support codehere  thats copied from the implementation, which in turn was copied from mu.py, i'd suggest dropping all of it, pre-creating the topic and unit testing just the message send.

Reply:
Ignoring the methods for a minute. I took a look at your aws test_sns_notify test and it looks like you follow the pattern I implemented.  You create the topic, execute the policy, then clean up(remove the topic).  Is your opinion that I not follow the same test method flow in this test? Happy to do it either way. Without topic creation in the test and with.

Reply:
a couple of things, one i assume your referencing a functional test, which is quite different in nature, moreover it does that setup with a line to create and and delete, here the test is dwarfed by the implementation to setup that infra. guiding principles should be around the least amount of code to do the job correctly, that makes it easier to maintain, change, and test. in this context there would be significant code reduction from removing the create/delete boilerplate around a topic, and just roll with a pre-created topic.

Reply:
Got you. Agreed. I have removed it in the upcoming commit.